runtime/doc/lsp.txt

```diff
@@ -2249,6 +2249,35 @@ stop({bufnr}, {client_id})                    *vim.lsp.semantic_tokens.stop()*
       • {client_id}  (`integer`) The ID of the |vim.lsp.Client|
 
 
+==============================================================================
+Lua module: vim.lsp.on_type_formatting                *lsp-on_type_formatting*
```

Comment:
I don't feel like this needs its own module. But if we don't want to add stuff to `lsp/buf.lua` then I would create a new `formatting.lua` module and move all the formatting handlers there, not make this on-type exclusive.

Reply:
Personally, `start_on_type_formatting` is too much of a mouthful, so I would prefer `lsp.formatting.enable()` (with proper options) here, mirroring the `lsp.completion` module. (Formatting seems to be on the same level as completion, conceptually.)

Reply:
Wouldn't `vim.lsp.formatting` be a little better. `completion` means to `complete`, and `formatting` means to `format`.